Enhance sol stable

RoyCai7 opened the pull request:
poo#[118381](https://progress.opensuse.org/issues/118381)
Sometimes ipmi sol does not work, reset it and re-send key 'up' again.

- Related ticket: [poo#118381](https://progress.opensuse.org/issues/118381)
- Needles:  
         - Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
         - Tag name: virttest-bootmenu-xen-kernel-count-down
- Verification run: 
         - [sle15sp4](http://openqa.qam.suse.cz/tests/48690)
         - [sle15sp4 Check_screen_count_down](http://openqa.qam.suse.cz/tests/48657#step/login_console/4) (just for test module)

RoyCai7 commented:
@alice-suse @Julie-CAO @waynechen55 @nanzhg @tbaev @guoxuguang @martinsmarcelo Welcome review, thanks.

Julie-CAO commented:
Anyway to make ipmi console work better is welcome. Could you have a verification run for product test as well, fg. https://openqa.suse.de/tests/10087544 with 'LTSS=0' settings in the command line? Ignore the test failures, just watch if login_console and reboot_and_wait_up_normal goes well.

alice-suse commented on tests/virt_autotest/login_console.pm:
```diff
@@ -120,6 +120,12 @@ sub login_to_console {
             send_key 'up';
             save_screenshot;
+            #If sol console is stuck, reconnect it and resend key up
+            if (check_screen 'virttest-bootmenu-xen-kernel-count-down', 1) {
```
I believe this is an effective way to proceed with xen grub menuentry selection when sol is not functioning well. Good job! 

Just a minor comment, the timeout is 1 second(so short)? Can it survive most such situations?

RoyCai7 replied:
In my test, it is enough and can survive.
also add it to 2 seconds, it's much enough.
It must be completed in count-down default 10 seconds.

alice-suse commented:
> * Needles:
>   - Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
>   - Tag name: virttest-bootmenu-xen-kernel-count-down

Can you provide the needle MR? The merge of this PR has to wait until that MR is merged.

alice-suse commented:
> * Verification run:
>   - [sle15sp4](http://openqa.qam.suse.cz/tests/48690)
>   - [sle15sp4 Check_screen_count_down](http://openqa.qam.suse.cz/tests/48657#step/login_console/4) (just for test module)

@Julie-CAO  From these verification runs, it seems your new PR of detecting coredump/calltrace is enabled in qem test too. It will be helpful if you can introduce to qem guys how it work , how to check the soft_failure and uploaded logs to open potential new bugs. Will it be okay for you to do a short introduction in today's sprint meeting?

RoyCai7 commented:
> > > * Verification run:
> > >   
> > >   * [sle15sp4](http://openqa.qam.suse.cz/tests/48690)
> > >   * [sle15sp4 Check_screen_count_down](http://openqa.qam.suse.cz/tests/48657#step/login_console/4) (just for test module)
> > 
> > 
> > @Julie-CAO From these verification runs, it seems your new PR of detecting coredump/calltrace is enabled in qem test too. It will be helpful if you can introduce to qem guys how it work , how to check the soft_failure and uploaded logs to open potential new bugs. Will it be okay for you to do a short introduction in today's sprint meeting?
> 
> There is a warning/softfail notice in the test. I'll remind them to be aware. Do they really need some introduction about how to check failure and upload logs? The journal check in the test code is very easy, just "journalctl | grep - i -e 'core dump' -e 'call trace'".

Please ignore 'call trace' issue. I reset server many times so it appear.

RoyCai7 commented:
> > * Needles:
> >   
> >   * Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
> >   * Tag name: virttest-bootmenu-xen-kernel-count-down
> 
> Can you provide the needle MR? The merge of this PR has to wait until that MR is merged.

I added it in OSD page directly.

Julie-CAO commented:
> Anyway to make ipmi console work better is welcome. Could you have a verification run for product test as well, fg. https://openqa.suse.de/tests/10087544 with 'LTSS=0' settings in the command line? Ignore the test failures, just watch if login_console and reboot_and_wait_up_normal goes well.

Sorry, it should be "LTSS=1'. If 'LTSS=0' has been used already, it is ok. It takes longer, you can cancel the test in any module after reboot_and_wait_up_normal.

alice-suse commented:
> > > * Needles:
> > >   
> > >   * Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
> > >   * Tag name: virttest-bootmenu-xen-kernel-count-down
> > 
> > 
> > Can you provide the needle MR? The merge of this PR has to wait until that MR is merged.
> 
> I added it in OSD page directly.

Do you mean openqa.suse.de? This is OSD actually.
But if you actually mean void.qam.suse.cz , I am afraid the needle is only added to void's needle directory. But you can double check if it exists in https://gitlab.suse.de/openqa/os-autoinst-needles-sles.git.

alice-suse commented:
> > > * Needles:
> > >   
> > >   * Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
> > >   * Tag name: virttest-bootmenu-xen-kernel-count-down
> > 
> > 
> > Can you provide the needle MR? The merge of this PR has to wait until that MR is merged.
> 
> I added it in OSD page directly.

Do you mean openqa.suse.de? This is OSD actually.
But if you actually mean void.qam.suse.cz , I am afraid the needle is only added to void's needle directory. But you can double check if it exists in https://gitlab.suse.de/openqa/os-autoinst-needles-sles.git.

RoyCai7 commented:
> > > > * Needles:
> > > >   
> > > >   * Needle name: login_console-virttest-bootmenu-xen-kernel-count-down-20221207
> > > >   * Tag name: virttest-bootmenu-xen-kernel-count-down
> > > 
> > > 
> > > Can you provide the needle MR? The merge of this PR has to wait until that MR is merged.
> > 
> > 
> > I added it in OSD page directly.
> 
> Do you mean openqa.suse.de? This is OSD actually. But if you actually mean void.qam.suse.cz , I am afraid the needle is only added to void's needle directory. But you can double check if it exists in https://gitlab.suse.de/openqa/os-autoinst-needles-sles.git.

It's my mistake, I added needle into void server directly.
I will check it in the link: https://gitlab.suse.de/openqa/os-autoinst-needles-sles.git
and submit a MR.

waynechen55 commented on tests/virt_autotest/login_console.pm:
```diff
@@ -120,6 +120,12 @@ sub login_to_console {
+                reset_consoles;
+                select_console 'sol', await_console => 0;
+                send_key 'up';
+            }
```
I think your change may have impact on other archs, for example, arm64 as well. It needs to be paid attention.

waynechen55 replied:
I do not think the new needle can cover suitable amount of qualified scenarios for you to reset console subsequently.  Maybe the new needle is not necessary at all.  

For example, the following code snippet can also serve your purpose:

```
if (is_xen_host) {
    #send key 'up' to stop grub timer counting down, to be more robust to select xen
            
    for (1 .. 20) {
        disable_grub_timeout_sol;
        select_grub_xen_sol; 
    }
}

sub disable_grub_timeout_sol {
    reset_consoles;
    select_console 'sol', await_console => 0;
    for (1 .. 5) {
        send_key 'up';
        save_screenshot;
    }
}

sub select_grub_xen_sol {
    reset_consoles;
    select_console 'sol', await_console => 0;
    for (1 .. 20) {
        send_key 'down';
        last if check_screen 'virttest-bootmenu-xen-kernel', 5;
    }
```
